apps/trading/hooks/use-orders.ts

```diff
@@ -0,0 +1,149 @@
+import { gql, useApolloClient } from '@apollo/client';
+import { useCallback, useEffect, useState } from 'react';
+import {
+  OrderSub,
+  OrderSubVariables,
+  Orders,
+  OrdersVariables,
+  OrderFields,
+} from '@vegaprotocol/graphql';
+import { singletonHook } from 'react-singleton-hook';
+import uniqBy from 'lodash.uniqby';
+import orderBy from 'lodash.orderby';
+import { useVegaWallet } from '@vegaprotocol/wallet';
+
+const ORDER_FRAGMENT = gql`
+  fragment OrderFields on Order {
+    id
+    market {
+      id
+      name
+      decimalPlaces
+      tradableInstrument {
+        instrument {
+          code
+        }
+      }
+    }
+    type
+    side
+    size
+    status
+    rejectionReason
+    price
+    timeInForce
+    remaining
+    expiresAt
+    createdAt
+    updatedAt
+  }
+`;
+
+export const ORDERS_QUERY = gql`
+  ${ORDER_FRAGMENT}
+  query Orders($partyId: ID!) {
+    party(id: $partyId) {
+      id
+      orders {
+        ...OrderFields
+      }
+    }
+  }
+`;
+
+export const ORDERS_SUB = gql`
+  ${ORDER_FRAGMENT}
+  subscription OrderSub($partyId: ID!) {
+    orders(partyId: $partyId) {
+      ...OrderFields
+    }
+  }
+`;
+
+interface UseOrders {
+  orders: OrderFields[];
+  error: Error | null;
+  loading: boolean;
+}
+
+export const useOrdersImpl = () => {
```

Comment:
What does Impl mean? can this just be useOrders?

Reply:
Impl here just means 'implementation'. This hook then gets passed to the `singletonHook` hook function whose result is named `useOrders` (see bottom of the file). This is so that there is only one instance of the hook and therefore one subscription started no matter how many times `useOrders` is used.

Also, `useOrdersImpl` is exported for testing purposes